JenkinsfileStressTests.groovy

```diff
@@ -117,6 +117,11 @@ pipeline {
                 }
             }
             stages {
+                stage('Gatling warmup') {
+                    steps {
+                        build job: 'Gatling-job', parameters: [[$class: 'StringParameterValue', name: 'SBT_ACTION', value: "gatling:testOnly ${params.SIMULATION}"], [$class: 'StringParameterValue', name: 'GITHUB', value: params.GITHUB_SIMULATIONS], [$class: 'StringParameterValue', name: 'DURATION', value: params.SIMULATION_DURATION]]
```

Comment:
why are you not using Gatling native warmup capabilities?

Reply:
Because we are not warming gatling up, but James!

https://gatling.io/docs/current/http/http_protocol/

```
Automatic warm up

The Java/NIO engine start up introduces an overhead on the first request to be executed. In order to compensate this effect, Gatling automatically performs a request to https://gatling.io.

To disable this feature, just add .disableWarmUp to an HTTP Protocol Configuration definition. To change the warm up url, just add .warmUp("newUrl").
```

Not what we want...

Reply:
`Then, FrontLine has warm-up and ramp-down phases exclusion built-in :)`

Not sure we wonna pay frontline...

https://groups.google.com/g/gatling/c/w2BsrnE1qxM `Is it possbile to warm up application under test?`

Reply:
We could rely on "Sliencing" https://gatling.io/docs/current/http/http_request/#silencing but it requires each simulation to handle that...

Having a systematic, gatling development transparent way to do it is appealing to me!

Reply:
ok for the way then, but why James would need to be warm up?? Even if there is a good reason, I don't see how you could achieve that in production.

Reply:
And I don't get neither why rampUsers() is not appropriate for this case

Reply:
> Even if there is a good reason, I don't see how you could achieve that in production.

As far as am I aware of, bench ci do not run against a production.

Also jit, caches are quitte tricky to warm, a get a really bad sens of measure on our "release" automated runs. I am unsure some ' generic warm up' would fit.

In the hanoi team, we tend tu run manually the gatling simulation, cancel, then do the relevant run.

Lets just automate that.

Reply:
Well "inject ar fixed rate' dont have a "but ramp before".

Reply:
> As far as am I aware of, bench ci do not run against a production.

I may have misspoken. I meant: if you need this for your test, what will you expect on production? To ask users to use slowly James when it starts? To ask admins to open slowly the service when it restarts?

Reply:
> Well "inject ar fixed rate' dont have a "but ramp before".

Injection can be chained, but maybe I don't get exactly what you mean.

Reply:
Even with a ramp, the few first request are still slow, even without load. Ramp imo is a less good solution than a warm up.

And it requires to rewrite all simulations.

And it would require every future simulation developer to be aware about it.

Reply:
> I may have misspoken. I meant: if you need this for your test, what will you expect on production? To ask users to use slowly James when it starts? To ask admins to open slowly the service when it restarts?

I guess when you start doesn't reflect what is your server is capable of when warm. It's not a problem with James, but Cassandra by the way. Cassandra is very slow to warm up. I feel it's slower in the provisioning case using BTRFS than the simulations with a fresh cassandra (I don't know if it's because of massive provisioning or BTRFS architecture)

On that... Chaining injections is not a bad idea, I guess you could start by a slow rampUp the time to warmup and then do your fixed rate injection, but then we would need to rewrite and retest all simulations...

Reply:
>  I feel it's slower in the provisioning case using BTRFS than the simulations with a fresh cassandra (I don't know if it's because of massive provisioning or BTRFS architecture)

Cache effect. The time for the OS to map all SSTable files in memory :-)

There's a bit more data with the provisionning!

Reply:
Also actually @rouazana you would not loose the data on warmup as it seems to be your concern. Might be even better? If I succeed you will get two gatling reports regarding the simulation: 
* one belonging to the warmup
* one belonging to the simulation after warmup

Then you can say we have those perfs during the warmup, and when everything is warm we can go up to those perfs... You might get more accurate values then, specifically on P99?

Would that make sense?

Reply:
If we put the warmup here, that means that the load results we will be promoting will be different from the results anyone can get by running loads on the james-gatling repository.

Reply:
No issue with me.

It promotes the internal usage of the bench ci, so...